### DISCUSSION TITLE
Mutate doesn't seem to be working

### DISCUSSION THREAD

**Discussion by Palmlight** [vercel/swr#1476] [Q&A] [answered]
I tried mutating a parent url 

eg:    url={`/users?type=${type}&paginate=10&page=${page}`}

in a child component by doing 

import { mutate } from "swr";

  mutate(url);

but it doesn't revalidate. 

Please am I doing something wrong?

I said the documentation is old because I also tried to use the documentation's 

import useSWR, { useSWRConfig } from 'swr'

function App () {
  const { mutate } = useSWRConfig()

but useSWRConfig doesn't seem to appear

**Comment by shuding** [comment, marked as answer]
`mutate` is working, we have tests covered. Which version of SWR are you currently using? Can you provide a bug reproduction on https://codesandbox.io?

**Reply by Palmlight** [reply]
I'm using 

    "swr": "^1.0.1",
I can't really reproduce on codesandbox as it's an internal api but the docs says to use useSWRConfig() which is not exported from swr. Then the mutate coming from swr doesn't work. I can show you a via a google meet or a zoom call

**Reply by shuding** [reply]
I just confirmed that `useSWRConfig` is correctly exported and `mutate` is working properly.

Can you at least create a reproduction on CodeSandbox to proof that "useSWRConfig() which is not exported from swr"? Happy to take a look if that's the case!

Furthermore, you can use a fake fetcher to reproduce any other issues, the actual API shouldn't matter.

**Reply by Palmlight** [reply]
Hi, useSWRConfig is working perfectly now. Thanks a lot but I have one more issue. If I have a url on a parent page and call mutate on the parent page. 

Eg: I call mutate on index.jsx it works but if i call the mutate from a component eg: IndexComponent.jsx it doesn't mutate. Do I create a mutate function on the parent and pass down to the child or use just the URL in the child?

**Reply by Palmlight** [reply]
Passing a mutate function down to the child works perfectly can you confirm if this is a good practice. Hope I'm not stressing you out with my questions?
🥺

**Reply by shuding** [reply]
https://swr.vercel.app/docs/mutation

There're two kinds of `mutate` functions, one is returned by `useSWR`, bound to the key. Another one is the global one, that you can pass a specific key to it. But it doesn't matter where you call the `mutate` functions from.
